src/index.js

```diff
@@ -120,7 +153,17 @@ const embedPublicKey = (publicKey, entry, callback) => {
  * @return {Void}
  */
 const extractPublicKey = (peerId, entry, callback) => {
-  callback(new Error('not implemented yet'))
+  if (entry.pubKey) {
+    try {
+      const pubKey = crypto.keys.unmarshalPublicKey(entry.pubKey)
+
+      return callback(null, pubKey)
```

Comment:
Avoid callback in a `try` block - if the callback throws, it'll be caught and the callback will be called again - very hard to debug!

Reply:
true!